backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/ClusterFactory.java

```diff
@@ -19,51 +19,260 @@
+    public ClusterFactory username(String username) {
+        this.username = Optional.of(username);
+
+        return this;
+    }
+
+    public ClusterFactory password(String password) {
+        this.password = Optional.of(password);
+
+        return this;
     }
 
-    public static Cluster createTestingCluster(String ip, int port) {
-        return createClusterForClusterWithPassWord(ImmutableList.of(new CassandraNodeIpAndPort(ip, port)), null, null, Optional.of(0));
+    public ClusterFactory host(String host) {
+        this.host = Optional.of(host);
+
+        return this;
+    }
+
+    public ClusterFactory port(int port) {
+        this.port = Optional.of(port);
+
+        return this;
+    }
+
+    public ClusterFactory refreshSchemaIntervalMillis(int refreshSchemaIntervalMillis) {
+        this.refreshSchemaIntervalMillis = Optional.of(refreshSchemaIntervalMillis);
+
+        return this;
     }
 
-    public static Cluster createDefaultSession() {
-        return createClusterForSingleServerWithoutPassWord(DEFAULT_CLUSTER_IP, CassandraNodeIpAndPort.DEFAULT_CASSANDRA_PORT);
+    public ClusterFactory servers(CassandraNodeIpAndPort ...servers) {
+        this.servers = Optional.of(ImmutableList.copyOf(servers));
+
+        return this;
+    }
+
+    public ClusterFactory servers(Collection<CassandraNodeIpAndPort> servers) {
+        this.servers = Optional.of(servers);
+
+        return this;
+    }
+
+    public ClusterFactory forTest() {
+        forTest = true;
+
+        return this;
+    }
+
+    public ClusterFactory queryLogger(QueryLoggerConf queryLogger) {
+        this.queryLogger = Optional.of(queryLogger);
+
+        return this;
+    }
+
+    public Cluster build() {
+        Preconditions.checkState(!(servers.isPresent() && host.isPresent()), "You can't specify a list of servers and a host at the same time");
+        Preconditions.checkState(!(servers.isPresent() && port.isPresent()), "You can't specify a list of servers and a port at the same time");
+        Preconditions.checkState(username.isPresent() == password.isPresent(), "If you specify username, you must specify password");
+        Preconditions.checkState(forTest == refreshSchemaIntervalMillis.isPresent(), "You can't specify refreshSchemaIntervalMillis for test");
+
+        Cluster.Builder clusterBuilder = Cluster.builder();
+        getServers().forEach(
+                (server) -> clusterBuilder.addContactPoint(server.getIp()).withPort(server.getPort())
+        );
+
+        username.map(username ->
+            password.map(password ->
+                clusterBuilder.withCredentials(username, password)
+            )
+        );
+
+        getRefreshSchemaIntervalMillis().map(refreshSchemaIntervalMillis ->
+            clusterBuilder.withQueryOptions(
+                new QueryOptions()
+                    .setRefreshSchemaIntervalMillis(refreshSchemaIntervalMillis)
+            )
+        );
+
```

Comment:
@aduprat If I do so then eclipse transform the following code:
```
        username.map(username ->
            password.map(password ->
                clusterBuilder.withCredentials(username, password)
            )
        );

        getRefreshSchemaIntervalMillis().map(refreshSchemaIntervalMillis ->
            clusterBuilder.withQueryOptions(
                new QueryOptions()
                    .setRefreshSchemaIntervalMillis(refreshSchemaIntervalMillis)
            )
        );

```

In:

```
        username.map(username -> password.map(password -> clusterBuilder.withCredentials(username, password)));

        getRefreshSchemaIntervalMillis().map(refreshSchemaIntervalMillis -> clusterBuilder.withQueryOptions(new QueryOptions().setRefreshSchemaIntervalMillis(refreshSchemaIntervalMillis)));
```

which is definitly worst and I know that if I push something like that @rouazana will tell me in the next minute add breakline! So do you really follow this file ? If yes why isn't in the git repository. If we want to impose a strict formatting why don't we create a editor config file that will work well both with Eclipse and IntelliJ.
If we go back to the original question, I guess you want me to for @chibenwa style ?